src/plugins/vis_augmenter/public/utils/utils.ts

```diff
@@ -58,3 +64,38 @@ export const buildPipelineFromAugmentVisSavedObjs = (objs: ISavedAugmentVis[]):
     throw new Error('Expression function from augment-vis saved objects could not be generated');
   }
 };
+
+/**
+ * Returns an error with an aggregated message about all of the
+ * errors found in the set of VisLayers. If no errors, returns undefined.
+ */
+export const getAnyErrors = (visLayers: VisLayer[], visTitle: string): Error | undefined => {
+  const visLayersWithErrors = visLayers.filter((visLayer) => isVisLayerWithError(visLayer));
+  if (!isEmpty(visLayersWithErrors)) {
+    // Aggregate by unique plugin resource type
+    const resourceTypes = [
+      ...new Set(visLayersWithErrors.map((visLayer) => visLayer.pluginResource.type)),
+    ];
+
+    let msgDetails = '';
+    resourceTypes.forEach((type, index) => {
+      const matchingVisLayers = visLayersWithErrors.filter(
+        (visLayer) => visLayer.pluginResource.type === type
+      );
+      if (index !== 0) msgDetails += '\n\n\n';
+      msgDetails += `-----${type}-----`;
```

Comment:
Why are we adding the `------` markers? Is this error user facing? If it is, i think we should not make it so cryptic and keep it clean and readable.

Reply:
It's for formatting the resource type header in the plaintext box within the modal. See screenshot in description for how it looks. This had been signed off from UX.